assets/app/components/site/siteBuildLogs.js

```diff
@@ -0,0 +1,65 @@
+import React from "react";
+import buildLogActions from "../../actions/buildLogActions";
+
+class SiteBuildLogs extends React.Component {
+  componentDidMount() {
```

Comment:
I'm curious about how to avoid triggering fetches in lifecycle callbacks?

For reference, a pattern I'm using to working in is [Presentation/Container components](https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0). I've used that pattern in this situation by having a stateful container component that triggered the fetch and then updates the props of stateless presentational components appropriately.

To avoid the situation where we display an empty state while waiting on data I'd configure the state to look like this:

```javascript
buildLogs: {
  state: "pending", // ["pending", "fulfilled", "rejected"],
  data: [],
  error: null,
}
```

When the `componentWillMount` method is called on the container I dispatch an action that will update the state to pending, trigger the fetch, and then dispatch and action to update the state to fulfilled / rejected with either data or an error. 

I didn't use that here because our state only seems to save the data fetched from the API and I wanted to stay consistent with the rest of the code in the frontend. 

The above pattern has worked for me, but still depends on lifecycle callbacks to trigger fetches, so I'm curious about other methods.